src/sql/src/pure.rs

```diff
@@ -111,8 +112,17 @@ pub async fn purify(mut stmt: Statement<Raw>) -> Result<Statement<Raw>, anyhow::
                 namespace,
                 table,
                 columns,
+                slot,
                 ..
-            } => purify_postgres_table(conn, namespace, table, columns).await?,
+            } => {
+                slot.get_or_insert_with(|| {
+                    format!(
+                        "materialize_{}",
+                        Uuid::new_v4().to_string().replace('-', "")
```

Comment:
why do we have to update the slot name?

Reply:
This only sets the slot name if none is provided, is this what you mean?

Reply:
ah, sorry yes!